enhancements/baremetal/continuous-hw-introspection.md

```diff
@@ -0,0 +1,114 @@
+baremetal hosts which potentially can be used to monitor the health of individual nodes and
+the deployment as a whole.
+
+This enhancement will not result in a new CRD or API to be written but will detail
+data that would become availabe in the Status field of the BareMetalHost CRD.
+
+## Release Signoff Checklist
+
+- [X] Enhancement is `implementable`
+- [X] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Continuous HW inspection on all the master and worker nodes that are a part of
+a Baremetal IPI deployment, allow the user to detect changes in server hardware
+and become aware of any alarms or failures seen on the servers. This information
+can be used to report important information to the user and/or glean metrics
+about the health and performance of the deployment.
+
+At the end of 4.4, a Baremetal IPI deployment does not have hardware introspection
+running on its master or worker nodes. The underlying provisioning service already has
+the capability to perform hardware inspection on each of the servers periodically and update
+the "status" field of the BareMetalHost CRD. Please refer to [1] for the latest BaremetalHost
+CRD.
+
+## Motivation
+
+This enhancement will allow the user to get continuous updates on all the baremetal
+servers in their Metal3 deployment providing useful metrics about the deployment and
+monitor the health of the baremetal servers.
+
+### Goals
+
+This is enhancement will allow the baremetal host CR associated with each baremetal
+server in a metal3 deployment to contain up-to-date informtaion about the HW and
+operational status of a server.
+
+### Non-Goals
+
+This proposal does not include any changes to the BareMetalHost CRD.
+
+## Proposal
+
+This enhancement proposes to add a new Daemonset started by the machine-api-operator
+only for the BareMetal platform. This Daemonset would run a new container image
+(ironic-hardware-inventory-recorder) on all master and worker nodes. This new service
+would keep the Hardware status on each of these baremetal servers updated by continuously
+gathering HW parameters on each of these baremetal nodes.
+
+### Implementation Details/Notes/Constraints [optional]
+
+The machine-api-operator would now be responsible for deploying the Daemonset that would
+run the service that performs continuous HW introspection on baremetal hosts.The container
+image for the Daemonset can be found at [2]. Also, if for some reason, a previously "ready"
+metal3 deployment becomes "unavailable", the daemonset can continue to keep running.
+This just will result in the lastest state of the baremetal servers in the baremetal CRs
+eventually become stale until metal3 transitions to "available"  operational state again.
```

Comment:
Should we include information about how baremetal-operator will use this information? e..g when it's not busy reconciling it should be polling Ironic to update the Status field with introspection data

Reply:
In the past, we have stayed away from mentioning Ironic in our proposals because it was considered an implementation detail. So, I'll try to word it without actually mentioning Ironic.